datadog_checks_base/datadog_checks/base/checks/win/wmi/__init__.py

```diff
@@ -226,48 +240,70 @@ def _submit_metrics(self, metrics, metric_name_and_type_by_property):
             func(metric_name, metric.value, metric.tags)
 
     def _get_instance_key(self, host, namespace, wmi_class, other=None):
+        # type: (str, str, str, Any) -> str
         """
         Return an index key for a given instance. Useful for caching.
         """
         if other:
             return "{host}:{namespace}:{wmi_class}-{other}".format(
                 host=host, namespace=namespace, wmi_class=wmi_class, other=other
             )
-
         return "{host}:{namespace}:{wmi_class}".format(host=host, namespace=namespace, wmi_class=wmi_class)
 
+    def get_running_wmi_sampler(self, properties, filters, **kwargs):
+        # type: (List[str], List[Dict[str, WMIFilter]], **Any) -> WMISampler
+        tag_by = kwargs.pop('tag_by', "")
+        return self._get_running_wmi_sampler(
+            instance_key=None,
+            wmi_class=self.wmi_class,
+            properties=properties,
+            filters=filters,
+            host=self.host,
+            namespace=self.namespace,
+            provider=self.provider,
+            username=self.username,
+            password=self.password,
+            tag_by=tag_by,
+            **kwargs
+        )
+
     def _get_running_wmi_sampler(self, instance_key, wmi_class, properties, tag_by="", **kwargs):
+        # type: (Any, str, List[str], str, Any) -> WMISampler
         """
         Return a running WMISampler for the given (class, properties).
 
         If no matching WMISampler is running yet, start one and cache it.
         """
-        properties = list(properties) + [tag_by] if tag_by else list(properties)
+        if not self._wmi_sampler:
+            property_list = list(properties) + [tag_by] if tag_by else list(properties)
+            self._wmi_sampler = WMISampler(self.log, wmi_class, property_list, **kwargs)
+            self._wmi_sampler.start()
 
-        if instance_key not in self.wmi_samplers:
-            wmi_sampler = WMISampler(self.log, wmi_class, properties, **kwargs)
-            wmi_sampler.start()
-            self.wmi_samplers[instance_key] = wmi_sampler
-
-        return self.wmi_samplers[instance_key]
+        return self._wmi_sampler
 
     def _get_wmi_properties(self, instance_key, metrics, tag_queries):
+        # type: (Any, List[List[str]], List[List[str]]) -> WMIProperties
         """
         Create and cache a (metric name, metric type) by WMI property map and a property list.
         """
-        if instance_key not in self.wmi_props:
+        if not self._wmi_props:
             metric_name_by_property = dict(
                 (wmi_property.lower(), (metric_name, metric_type)) for wmi_property, metric_name, metric_type in metrics
-            )
-            properties = map(lambda x: x[0], metrics + tag_queries)
-            self.wmi_props[instance_key] = (metric_name_by_property, properties)
+            )  # type: Dict[str, Tuple[str, str]]
+            properties = map(lambda x: x[0], metrics + tag_queries)  # type: List[str]
```

Comment:
If we're not changing the behavior yet (which I agree with) we should probably keep `Iterator` - `List` here looks wrong (does mypy event accept this?)

Reply:
Yes, in fact if I say it's a Iterator I get this error:
```
datadog_checks/base/checks/win/wmi/__init__.py:293:26: note: 'list' is missing following 'Iterator' protocol member:
datadog_checks/base/checks/win/wmi/__init__.py:293:26: note:     next
```

Reply:
Oh, right it works (and with `type: Iterator` it doesn't) because `map` returns lists on Python 2, and mypy runs in Python 2 mode… erk. Okay.